src/x/time/time_test.go

```diff
@@ -161,3 +159,45 @@ func TestMaxTime(t *testing.T) {
 		require.Equal(t, input.expected, MaxTime(input.t1, input.t2))
 	}
 }
+
+func TestTruncate(t *testing.T) {
+	var (
+		n      = time.Now().UTC()
```

Comment:
nit: Perhaps best to set this to a static date? I wonder if it would flake if it's always "now".

Reply:
Do we have any way of marking a flake as not a flake 😛 ? If this flakes then it implies there's still some underlying problem with truncation, would be useful to fail out if at all possible